mlir/include/mlir/Dialect/Mesh/IR/MeshOps.td

```diff
@@ -196,16 +197,18 @@ def Mesh_ShardingOp : Mesh_Op<"sharding", [
     e.g. the first sharded dimension gets [1,2] halos and the seconds gets [2,3] halos.
     `?` indicates dynamic halo sizes.
     
-    6. [Optional] Sizes of sharded dimensions of each shard.
-    `sharded_dims_sizes`is provided as a flattened 1d array of i64s: for each device of the
-    device-mesh one value for each sharded tensor dimension.
+    6. [Optional] Offsets for each shard and sharded tensor dimension.
+    `sharded_dims_offsets` is provided as a flattened 1d array of i64s:
+    For each sharded tensor dimension the offsets (starting index) of all shards in that dimension.
+    The offset of each first shard is omitted and is implicitly assumed to be 0.
```

Comment:
I feel pretty strongly about having the offset computed as exclusive prefix sum of the "size" arrays. That is, the array is still size N, the first element is 0, and the offset for shard `i` can be retrieved as `sharded_dims_offsets[i]`.

Reply:
We will need an additional last element, otherwise we cannot deduce the size of of the last shard. 
That is because there is no way to implicitly define the global tensor shape. It should be possible to deduce the sizes without pulling in a dependency to the input tensor.  The start will always be 0, so that is redundant information and can be avoided to keep the array to size N. Not so the end.
I will change to having offset for shard `i` as `sharded_dims_offsets[i]` and append the offset to the end.

Reply:
done

Reply:
Apologies if i did not provide more context in my previous message, but is there an instance where the input tensor shape is not directly available that warrant the N+1 representation? 

If that's the case, then I would agree with you in that the original implementation (size N, inclusive prefix sum representing the end offset) was probably better, especially because now you have a flattened 2d array of n+1 values which adds additional complexity.

Reply:
I see, at this point I don't know about a case where the shape is unavailable.

Still, I prefer a form that is self-sufficient. There is some complexity involved no matter what. If extracting the tensor size from another entity is more complex than dealing with a `N+1` values probably lies in the eye of the beholder.